components/datetime/src/options/length.rs

```diff
@@ -98,6 +99,27 @@ impl Default for Bag {
     }
 }
 
+impl Bag {
+    /// Constructs a Bag with all fields set to None
+    ///
+    /// Note that the [`Default`] implementation returns long date and time options
+    pub fn empty() -> Self {
+        Self {
+            date: None,
+            time: None,
+            preferences: None,
+        }
+    }
+
+    /// Constructs a Bag given a date and time field (preferences set to None)
+    pub fn from_date_time_style(date: Option<Date>, time: Option<Time>) -> Self {
```

Comment:
Suggestion: make the arguments non-optional here.

If `None` is ever required, it should be its own constructor (or manually configured).